File: string/count-the-number-of-vowel-strings-in-range/count-the-number-of-vowel-strings-in-range.ts
function vowelStrings(words: string[], left: number, right: number): number {
    const vowels = new Set(['a', 'e', 'i', 'o', 'u']);
    let count = 0;
    for (let i = left; i <= right; i++) {
        if (vowels.has(words[i][0]) && vowels.has(words[i][words[i].length - 1])) {
            count++;
        }
    }

    return count;
};

File: string/count-the-number-of-vowel-strings-in-range/count-the-number-of-vowel-strings-in-range.go
package countthenumberofvowelstringsinrange

func vowelStrings(words []string, left int, right int) int {
	vowels := map[byte]struct{}{
		'a': {},
		'e': {},
		'i': {},
		'o': {},
		'u': {},
	}

	count := 0
	for i := left; i <= right; i++ {
		_, oks := vowels[words[i][0]]
		_, oke := vowels[words[i][len(words[i])-1]]

		if oks && oke {
			count++
		}
	}

	return count
}
